src/test/java/io/vertx/test/core/NetClientTLSConnectTest.java

```diff
@@ -0,0 +1,72 @@
+package io.vertx.test.core;
+
+import java.util.concurrent.CountDownLatch;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.core.net.JksOptions;
+import io.vertx.core.net.NetClient;
+import io.vertx.core.net.NetClientOptions;
+import io.vertx.core.net.NetSocket;
+
+/**
+ * @author <a href="http://oss.lehmann.cx/">Alexander Lehmann</a>
+ *
+ */
+public class NetClientTLSConnectTest extends VertxTestBase {
+
+  private HttpServer server;
+
+  @Before
+  public void startServer() throws InterruptedException {
+    CountDownLatch latch = new CountDownLatch(1);
+
+    HttpServerOptions serverOptions = new HttpServerOptions();
```

Comment:
it doesn't, I started testing the issue by connecting to https://www.yahoo.com/ and then wrote the local server as https, it should work with a NetServer as well